### PR TITLE
Testing/teacher lessons

### LINKED ISSUE #79: TeacherLessons Testing

### PR DESCRIPTION
**Pull request by Melizzo**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures
## Pull request type
 
 
Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe): 
## What is the current behavior?

- TeacherLessons was not tested, a test was missing for lessonCard and proptypes were missing
## What is the new behavior?

- Completed testing for TeacherLessons (100% fns & statements) added 1 test for lessonCard(100% fns & statements

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

closes #77
closes #79
Screen Shot 2020-09-13 at 5 30 30 PM

### CODE REVIEW

File: src/LessonDetails/LessonDetails.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import PropTypes from 'prop-types'
 const LessonDetails = ({ lesson }) => {
 return(
   <section>
-    <h3>{`${lesson.name}`}</h3>
+    <h3>{lesson.name}</h3>
     <p>Scores will be shown here</p>
```
Comment (Melizzo) [inline src/LessonDetails/LessonDetails.js:8]: Thanks for the refactor Kyle

File: src/TeacherLessons/LessonCard/LessonCard.test.js [hunk 2]
```diff
@@ -23,4 +23,18 @@ describe('LessonCard', ()=> {
...
+      <BrowserRouter>
+          <LessonCard lessonTitle={mockLesson.name}
+          findLesson={mockToggle}/>
+      </BrowserRouter>
+    )
+
+    const titleToClick = getByText('Mocked Lesson')
+    fireEvent.click(titleToClick)
+    expect(mockToggle).toHaveBeenCalled()
+  })
```
Comment (Melizzo) [inline src/TeacherLessons/LessonCard/LessonCard.test.js:39]: thanks Bill for reminding me to expect the mocked funtion

File: src/TeacherLessons/TeacherLessons.js [hunk 3]
```diff
@@ -27,31 +26,16 @@ const TeacherLessons = ({ students, lessons }) => {
...
-
-  // State of the lessons added from Redux? Or delete from store via redux
-  // const deleteLesson = (e) => {
-  //   //eslint-disable-next-line
-  //   setLessons(lessons.filter(lesson => lesson.id != e.target.parentNode.id))
-  // }
-
-  // eslint-disable-next-line
-  let lessonCard;
+
```
Comment (Melizzo) [inline src/TeacherLessons/TeacherLessons.js:29]: Let's clean up some unneeded notes :)

File: src/mockData/mockData.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 export const lesson = {
   name: 'Test Title',
+  id: 123123,
   questions: [
```
Comment (Melizzo) [inline src/mockData/mockData.js:4]: This was a cause of some hurdles, but I cleared them!

File: src/TeacherLessons/TeacherLessons.js [hunk 5]
```diff
@@ -27,31 +26,16 @@ const TeacherLessons = ({ students, lessons }) => {
       )
     } 
   }
-
-  // Upon load of test Teacher:
```
Comment (Billwilke42) [inline src/TeacherLessons/TeacherLessons.js:31]: I'll rubber-duck this with you tomorrow. We were close.

File: src/TeacherLessons/TeaherLessons.test.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
+import React from 'react';
```
Comment (Billwilke42) [inline src/TeacherLessons/TeaherLessons.test.js:1]: Crush mode! Well done

**Review by Billwilke42 — APPROVED** [review]
Nice work Melissa!

File: src/mockData/mockData.js (hunk 4 above)
Comment (KyleWong2510) [reply src/mockData/mockData.js:4]: That would do it!  Nice job finding that.

**Review by KyleWong2510 — APPROVED** [review]
Good job on this stuff!

**Pull request merged**